server/src/main/java/com/cloud/network/element/ConfigDriveNetworkElement.java

```diff
@@ -254,10 +254,6 @@ public boolean saveSSHKey(final Network network, final NicProfile nic, final Vir
 
         final boolean canHandle = canHandle(network.getTrafficType());
 
-        if (canHandle) {
```

Comment:
since password is not reset when reset sshkey (in #4819 ), it is not needed to save password in vm details.